package/MDAnalysis/analysis/density.py

```diff
@@ -400,7 +400,8 @@ def __init__(self, atomgroup, delta=1.0,
 
     def _prepare(self):
         coord = self._atomgroup.positions
-        if self._gridcenter is not None:
+        if (self._gridcenter is not None or
```

Comment:
So now setting any of the values triggers this path. I think that is a good change and should be the expected behaviour. @IAlibay you happy with this?


We should also confirm that no changes need to be made in the docs to reflect this.

Reply:
I think removing the mention that xdim, ydim and zdim if gridenter is `None` would be enough. Instead of just ignoring it, now an error will be thrown.

Reply:
LGTM